### ISSUE TITLE
rust 1.53.0 can't bootstrap rust 1.61.0

### ISSUE THREAD

**Issue by nbriggs** [oracle/solaris-userland#13]
The previous rust version distributed in the Solaris Userland git repository was 1.53.0, on 26 July 2021.
The current rust version in Userland is 1.61.0 distributed 26 May 2022.

Version 1.53.0 is not capable of compiling 1.61.0, and there doesn't appear to have been a binary package update that provides version 1.61.0, or any version that is capable of compiling 1.61.0.

Version 1.53.0 is also not new enough to compile the current version of Firefox that is being distributed in Userland.

Is there a plan to address this?

**Comment by alanc** [comment]
Yes, the plan is for the binary packages built from the rust 1.61.0 bits you see in the git repository to be delivered in the Solaris 11.4.51 package repository.

**Comment by nbriggs** [comment]
Thanks Alan, that's good news, I guess.

Since the binary package builds seem to be pretty slow in coming it sure would be nice if either (a) one could discern a date when new binary packages might be released, or (b) when things are committed to the git repository one could build them based on having built (or got the binary packages for...) the previous contents of the git repository.  Having to wait for Oracle to build it all rather defeats the (perceived, from the outside) purpose of having the repository, doesn't it?

**Comment by nbriggs** [comment]
I've worked forward from 1.53.0 to 1.57.0 by back-porting the patches from Userland's rust 1.61.0, using the generated manifests in place of the .p5m files, and building each release one step at a time.

The build process is disturbingly unreliable, as evidenced by the build script's "try it three times to see if it works" being hard-coded in, and actually observing it having to do this.  The non-repeatability does not inspire great confidence in the system though I'm not sure if it's Solaris or Rust that is misbehaving.

**Comment by nbriggs** [comment]
Re: "Yes, the plan is for the binary packages built from the rust 1.61.0 bits you see in the git repository to be delivered in the Solaris 11.4.51 package repository."

Should we expect a new CBE release anytime soon?  Current CBE still seems to be 11.4.42.111.0, and it's not cost-effective to pay for an Oracle support contract to get SRU releases for non-commercial hobbyist usage.

**Comment by alanc** [comment]
While we are planning on another CBE release, we don't have a schedule for it yet.

**Comment by alanc** [comment]
> While we are planning on another CBE release, we don't have a schedule for it yet.

It took a bit longer than planned, but we shipped another CBE release last week:
https://blogs.oracle.com/solaris/post/whats-new-in-the-oracle-solaris-11481-cbe-release

**Comment by nbriggs** [comment]
@alanc That's fantastic.  Thanks for updating/closing this issue.
